### PR TITLE
Task 90: Share code api implementation

### PR DESCRIPTION
**Pull request by sorinburghiu2323**
Get and generate a share code.
To be handled in the frontend as per documentation.

### CODE REVIEW

File: backend/Controllers/user_controller.py [hunk 1]
```diff
@@ -205,12 +208,28 @@ def get_user(request, user_id):
...
         )
+
+    # Check for share code.
+    valid_sc = False
+    if get_user.share_code:
+        if request.GET.get("sharecode") == get_user.share_code:
+            valid_sc = True
+
+    if not valid_sc:
+        try:
```
Comment (milokhiggins) [inline backend/Controllers/user_controller.py:226]: This doesn't work 100%; I've tested it and when logged in, you can get the profile even if the code is invalid (or not given, same thing essentially).

**Review by milokhiggins — CHANGES_REQUESTED** [review]
looks good, you just need to fix the profile view. See my comment for details.

File: backend/Controllers/user_controller.py (hunk 1 above)
Comment (sorinburghiu2323) [reply backend/Controllers/user_controller.py:226]: We agreed that any logged in user should be able to see another user's profile. If you dont have an account/dont want to make one, then you would need the share code.

Comment (milokhiggins) [reply backend/Controllers/user_controller.py:226]: Oh right I see. All good then

**Pull request merged**